Flink Tests failure :  java.lang.NoClassDefFoundError: Could not initialize class org.apache.beam.runners.core.construction.SerializablePipelineOptions

This flink load tests are affected by this error 
- beam_LoadTests_Go_ParDo_Flink_Batch (FAILED)
\-- beam_LoadTests_Go_SideInput_Flink_Batch (FAILED)
\-- beam_LoadTests_Python_Combine_Flink_Batch (FAILED)
\-- beam_LoadTests_Python_Combine_Flink_Streaming (FAILED)
\-- beam_LoadTests_Python_ParDo_Flink_Batch (FAILED)
\-- beam_LoadTests_Python_ParDo_Flink_Streaming (FAILED)

 

Stacktrace of the error 


14:03:35 root_transform_ids: "e11"
14:03:35 root_transform_ids: "e12"
14:03:35 root_transform_ids: "e13"
14:03:35 root_transform_ids: "e14"
14:03:35 requirements: "beam:requirement:org.apache.beam:pardo:splittable_dofn:v1"
14:03:35 2022/03/07 20:03:33 Prepared job with id: load-tests-go-flink-batch-pardo-1-0307182650_12830e27-009a-44e5-8390-a53ce5f264f0 and staging token: load-tests-go-flink-batch-pardo-1-0307182650_12830e27-009a-44e5-8390-a53ce5f264f0
14:03:35 2022/03/07 20:03:33 Using specified worker binary: 'linux_amd64/pardo'
14:03:46 2022/03/07 20:03:46 Staged binary artifact with token: 
14:03:47 2022/03/07 20:03:47 Submitted job: load0tests0go0flink0batch0pardo0100307182650-root-0307200346-d0077c8_166b7cfd-0805-481a-9a6a-52ba9bfddcbc
14:03:47 2022/03/07 20:03:47 Job state: STOPPED
14:03:47 2022/03/07 20:03:47 Job state: STARTING
14:03:47 2022/03/07 20:03:47 Job state: RUNNING
14:04:04 2022/03/07 20:04:04  (): org.apache.flink.client.program.ProgramInvocationException: Job failed (JobID: a8fc45aac0e16ee159027398311e082e)
14:04:04 	at org.apache.flink.client.deployment.ClusterClientJobClientAdapter.lambda$null$6(ClusterClientJobClientAdapter.java:125)
14:04:04 	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:616)
14:04:04 	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:591)
14:04:04 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
14:04:04 	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1975)
14:04:04 	at org.apache.flink.client.program.rest.RestClusterClient.lambda$pollResourceAsync$22(RestClusterClient.java:665)
14:04:04 	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:774)
14:04:04 	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:750)
14:04:04 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
14:04:04 	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1975)
14:04:04 	at org.apache.flink.runtime.concurrent.FutureUtils.lambda$retryOperationWithDelay$9(FutureUtils.java:394)
14:04:04 	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:774)
14:04:04 	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:750)
14:04:04 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
14:04:04 	at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:575)
14:04:04 	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:943)
14:04:04 	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:456)
14:04:04 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
14:04:04 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
14:04:04 	at java.lang.Thread.run(Thread.java:748)
14:04:04 Caused by: org.apache.flink.runtime.client.JobExecutionException: Job execution failed.
14:04:04 	at org.apache.flink.runtime.jobmaster.JobResult.toJobExecutionResult(JobResult.java:144)
14:04:04 	at org.apache.flink.client.deployment.ClusterClientJobClientAdapter.lambda$null$6(ClusterClientJobClientAdapter.java:123)
14:04:04 	... 19 more
14:04:04 Caused by: org.apache.flink.runtime.JobException: Recovery is suppressed by NoRestartBackoffTimeStrategy
14:04:04 	at org.apache.flink.runtime.executiongraph.failover.flip1.ExecutionFailureHandler.handleFailure(ExecutionFailureHandler.java:118)
14:04:04 	at org.apache.flink.runtime.executiongraph.failover.flip1.ExecutionFailureHandler.getFailureHandlingResult(ExecutionFailureHandler.java:80)
14:04:04 	at org.apache.flink.runtime.scheduler.DefaultScheduler.handleTaskFailure(DefaultScheduler.java:233)
14:04:04 	at org.apache.flink.runtime.scheduler.DefaultScheduler.maybeHandleTaskFailure(DefaultScheduler.java:224)
14:04:04 	at org.apache.flink.runtime.scheduler.DefaultScheduler.updateTaskExecutionStateInternal(DefaultScheduler.java:215)
14:04:04 	at org.apache.flink.runtime.scheduler.SchedulerBase.updateTaskExecutionState(SchedulerBase.java:666)
14:04:04 	at org.apache.flink.runtime.scheduler.SchedulerNG.updateTaskExecutionState(SchedulerNG.java:89)
14:04:04 	at org.apache.flink.runtime.jobmaster.JobMaster.updateTaskExecutionState(JobMaster.java:446)
14:04:04 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
14:04:04 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
14:04:04 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
14:04:04 	at java.lang.reflect.Method.invoke(Method.java:498)
14:04:04 	at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.handleRpcInvocation(AkkaRpcActor.java:305)
14:04:04 	at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.handleRpcMessage(AkkaRpcActor.java:212)
14:04:04 	at org.apache.flink.runtime.rpc.akka.FencedAkkaRpcActor.handleRpcMessage(FencedAkkaRpcActor.java:77)
14:04:04 	at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.handleMessage(AkkaRpcActor.java:158)
14:04:04 	at akka.japi.pf.UnitCaseStatement.apply(CaseStatements.scala:26)
14:04:04 	at akka.japi.pf.UnitCaseStatement.apply(CaseStatements.scala:21)
14:04:04 	at scala.PartialFunction.applyOrElse(PartialFunction.scala:127)
14:04:04 	at scala.PartialFunction.applyOrElse$(PartialFunction.scala:126)
14:04:04 	at akka.japi.pf.UnitCaseStatement.applyOrElse(CaseStatements.scala:21)
14:04:04 	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:175)
14:04:04 	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:176)
14:04:04 	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:176)
14:04:04 	at akka.actor.Actor.aroundReceive(Actor.scala:517)
14:04:04 	at akka.actor.Actor.aroundReceive$(Actor.scala:515)
14:04:04 	at akka.actor.AbstractActor.aroundReceive(AbstractActor.scala:225)
14:04:04 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:592)
14:04:04 	at akka.actor.ActorCell.invoke(ActorCell.scala:561)
14:04:04 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258)
14:04:04 	at akka.dispatch.Mailbox.run(Mailbox.scala:225)
14:04:04 	at akka.dispatch.Mailbox.exec(Mailbox.scala:235)
14:04:04 	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
14:04:04 	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
14:04:04 	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
14:04:04 	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
14:04:04 Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.apache.beam.runners.core.construction.SerializablePipelineOptions
14:04:04 	at java.io.ObjectStreamClass.hasStaticInitializer(Native Method)
14:04:04 	at java.io.ObjectStreamClass.computeDefaultSUID(ObjectStreamClass.java:1955)
14:04:04 	at java.io.ObjectStreamClass.access$100(ObjectStreamClass.java:79)
14:04:04 	at java.io.ObjectStreamClass$2.run(ObjectStreamClass.java:275)
14:04:04 	at java.io.ObjectStreamClass$2.run(ObjectStreamClass.java:273)
14:04:04 	at java.security.AccessController.doPrivileged(Native Method)
14:04:04 	at java.io.ObjectStreamClass.getSerialVersionUID(ObjectStreamClass.java:272)
14:04:04 	at java.io.ObjectStreamClass.initNonProxy(ObjectStreamClass.java:694)
14:04:04 	at java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2005)
14:04:04 	at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1852)
14:04:04 	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2186)
14:04:04 	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1669)
14:04:04 	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2431)
14:04:04 	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2355)
14:04:04 	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2213)
14:04:04 	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1669)
14:04:04 	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:503)
14:04:04 	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:461)
14:04:04 	at org.apache.flink.util.InstantiationUtil.deserializeObject(InstantiationUtil.java:615)
14:04:04 	at org.apache.flink.util.InstantiationUtil.deserializeObject(InstantiationUtil.java:600)
14:04:04 	at org.apache.flink.util.InstantiationUtil.deserializeObject(InstantiationUtil.java:587)
14:04:04 	at org.apache.flink.util.InstantiationUtil.readObjectFromConfig(InstantiationUtil.java:541)
14:04:04 	at org.apache.flink.api.java.typeutils.runtime.RuntimeSerializerFactory.readParametersFromConfig(RuntimeSerializerFactory.java:78)
14:04:04 	at org.apache.flink.runtime.operators.util.TaskConfig.getTypeSerializerFactory(TaskConfig.java:1246)
14:04:04 	at org.apache.flink.runtime.operators.util.TaskConfig.getOutputSerializer(TaskConfig.java:599)
14:04:04 	at org.apache.flink.runtime.operators.DataSourceTask.initInputFormat(DataSourceTask.java:324)
14:04:04 	at org.apache.flink.runtime.operators.DataSourceTask.invoke(DataSourceTask.java:106)
14:04:04 	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:758)
14:04:04 	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:573)
14:04:04 	at java.lang.Thread.run(Thread.java:750)
14:04:04 2022/03/07 20:04:04  (): java.lang.NoClassDefFoundError: Could not initialize class org.apache.beam.runners.core.construction.SerializablePipelineOptions
14:04:05 2022/03/07 20:04:05 Job state: FAILED
14:04:05 2022/03/07 20:04:05 Failed to execute job: job load0tests0go0flink0batch0pardo0100307182650-root-0307200346-d0077c8_166b7cfd-0805-481a-9a6a-52ba9bfddcbc failed
14:04:05 panic: Failed to execute job: job load0tests0go0flink0batch0pardo0100307182650-root-0307200346-d0077c8_166b7cfd-0805-481a-9a6a-52ba9bfddcbc failed
14:04:05 
14:04:05 goroutine 1 [running]:
14:04:05 github.com/apache/beam/sdks/v2/go/pkg/beam/log.Fatalf(0x1143648, 0xc000120000, 0x10323eb, 0x19, 0xc0003cde00, 0x1, 0x1)
14:04:05 	/var/jenkins_real_home/workspace/beam_LoadTests_Go_ParDo_Flink_Batch/src/sdks/go/pkg/beam/log/log.go:153 ****0xec
14:04:05 main.main()
14:04:05 	/var/jenkins_real_home/workspace/beam_LoadTests_Go_ParDo_Flink_Batch/src/sdks/go/test/load/pardo/pardo.go:105 ****0x3ca
14:04:05 
14:04:05 \> Task :sdks:go:test:load:run FAILED

Imported from Jira [BEAM-14174](https://issues.apache.org/jira/browse/BEAM-14174). Original Jira may contain additional context.
Reported by: andoni.guzman.

The original failure was due to that the job server artifact of flink 1.12 has long stopped updating (https://pantheon.corp.google.com/gcr/images/apache-beam-testing/global/beam_portability/beam_flink1.12_job_server). Switched to using Flink 1.13, now all jobs are having same error:

```
04:43:52 2022/10/21 08:43:52 Submitted job: load0tests0go0flink0batch0combine0101021065324-root-1021084351-5f971342_60528140-7d74-4569-a5ec-99335a2f7dfe
04:43:52 2022/10/21 08:43:52 Job state: STOPPED
04:43:52 2022/10/21 08:43:52 Job state: STARTING
04:43:52 2022/10/21 08:43:52 Job state: RUNNING
04:45:01 2022/10/21 08:45:01  (): java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.RuntimeException: Error while waiting for job to be initialized
04:45:01 	at org.apache.flink.util.ExceptionUtils.rethrow(ExceptionUtils.java:316)
04:45:01 	at org.apache.flink.api.java.ExecutionEnvironment.executeAsync(ExecutionEnvironment.java:1061)
04:45:01 	at org.apache.flink.api.java.ExecutionEnvironment.execute(ExecutionEnvironment.java:958)
04:45:01 	at org.apache.beam.runners.flink.FlinkBatchPortablePipelineTranslator$BatchTranslationContext.execute(FlinkBatchPortablePipelineTranslator.java:195)
04:45:01 	at org.apache.beam.runners.flink.FlinkPipelineRunner.runPipelineWithTranslator(FlinkPipelineRunner.java:132)
04:45:01 	at org.apache.beam.runners.flink.FlinkPipelineRunner.run(FlinkPipelineRunner.java:99)
04:45:01 	at org.apache.beam.runners.jobsubmission.JobInvocation.runPipeline(JobInvocation.java:86)
04:45:01 	at org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
04:45:01 	at org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
04:45:01 	at org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
04:45:01 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
04:45:01 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
04:45:01 	at java.lang.Thread.run(Thread.java:750)
04:45:01 Caused by: java.util.concurrent.ExecutionException: java.lang.RuntimeException: Error while waiting for job to be initialized
04:45:01 	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
04:45:01 	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1908)
04:45:01 	at org.apache.flink.api.java.ExecutionEnvironment.executeAsync(ExecutionEnvironment.java:1056)
04:45:01 	... 11 more
04:45:01 Caused by: java.lang.RuntimeException: Error while waiting for job to be initialized
04:45:01 	at org.apache.flink.client.ClientUtils.waitUntilJobInitializationFinished(ClientUtils.java:160)
04:45:01 	at org.apache.flink.client.deployment.executors.AbstractSessionClusterExecutor.lambda$execute$2(AbstractSessionClusterExecutor.java:82)
04:45:01 	at org.apache.flink.util.function.FunctionUtils.lambda$uncheckedFunction$2(FunctionUtils.java:73)
04:45:01 	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:616)
04:45:01 	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:591)
04:45:01 	at java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:457)
04:45:01 	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
04:45:01 	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
04:45:01 	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
04:45:01 	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)
04:45:01 Caused by: java.util.concurrent.ExecutionException: org.apache.flink.runtime.concurrent.FutureUtils$RetryException: Could not complete the operation. Number of retries has been exhausted.
04:45:01 	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
04:45:01 	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1908)
04:45:01 	at org.apache.flink.client.deployment.executors.AbstractSessionClusterExecutor.lambda$null$0(AbstractSessionClusterExecutor.java:83)
04:45:01 	at org.apache.flink.client.ClientUtils.waitUntilJobInitializationFinished(ClientUtils.java:140)
04:45:01 	... 9 more
04:45:01 Caused by: org.apache.flink.runtime.concurrent.FutureUtils$RetryException: Could not complete the operation. Number of retries has been exhausted.
04:45:01 	at org.apache.flink.runtime.concurrent.FutureUtils.lambda$retryOperationWithDelay$9(FutureUtils.java:386)
04:45:01 	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:774)
04:45:01 	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:750)
04:45:01 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
04:45:01 	at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:575)
04:45:01 	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:943)
04:45:01 	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:456)
04:45:01 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
04:45:01 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
04:45:01 	at java.lang.Thread.run(Thread.java:750)
04:45:01 Caused by: java.util.concurrent.CompletionException: org.apache.flink.runtime.rest.util.RestClientException: Response was neither of the expected type([simple type, class org.apache.flink.runtime.rest.messages.job.JobDetailsInfo]) nor an error.
04:45:01 	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326)
04:45:01 	at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338)
04:45:01 	at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:925)
04:45:01 	at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:967)
04:45:01 	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:940)
04:45:01 	... 4 more
04:45:01 Caused by: org.apache.flink.runtime.rest.util.RestClientException: Response was neither of the expected type([simple type, class org.apache.flink.runtime.rest.messages.job.JobDetailsInfo]) nor an error.
04:45:01 	at org.apache.flink.runtime.rest.RestClient.parseResponse(RestClient.java:502)
04:45:01 	at org.apache.flink.runtime.rest.RestClient.lambda$submitRequest$3(RestClient.java:466)
04:45:01 	at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:966)
04:45:01 	... 5 more
04:45:01 Caused by: org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot map `null` into type `long` (set DeserializationConfig.DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES to 'false' to allow)
04:45:01  at [Source: UNKNOWN; line: -1, column: -1] (through reference chain: org.apache.flink.runtime.rest.messages.job.JobDetailsInfo["maxParallelism"])
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:63)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1575)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.deser.std.NumberDeserializers$PrimitiveOrWrapperDeserializer.getNullValue(NumberDeserializers.java:176)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.deser.impl.PropertyValueBuffer._findMissing(PropertyValueBuffer.java:204)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.deser.impl.PropertyValueBuffer.getParameters(PropertyValueBuffer.java:160)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.deser.ValueInstantiator.createFromObjectWith(ValueInstantiator.java:288)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:202)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:520)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1390)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:362)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:195)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:322)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:4569)
04:45:01 	at org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2867)
04:45:01 	at org.apache.flink.runtime.rest.RestClient.parseResponse(RestClient.java:475)
04:45:01 	... 7 more
04:45:01 2022/10/21 08:45:01  (): org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot map `null` into type `long` (set DeserializationConfig.DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES to 'false' to allow)
04:45:01  at [Source: UNKNOWN; line: -1, column: -1] (through reference chain: org.apache.flink.runtime.rest.messages.job.JobDetailsInfo["maxParallelism"])
04:45:01 2022/10/21 08:45:01 Job state: FAILED
04:45:02 2022/10/21 08:45:01 Failed to execute job: job load0tests0go0flink0batch0combine0101021065324-root-1021084351-5f971342_60528140-7d74-4569-a5ec-99335a2f7dfe failed

```

--- a/.test-infra/dataproc/flink_cluster.sh
+++ b/.test-infra/dataproc/flink_cluster.sh
@@ -17,7 +17,7 @@
 #    Provide the following environment to run this script:
 #
 #    GCLOUD_ZONE: Google cloud zone. Optional. Default: "us-central1-a"
-#    DATAPROC_VERSION: Dataproc version. Optional. Default: 1.5
+#    DATAPROC_VERSION: Dataproc version. Optional. Default: 2.1
 #    CLUSTER_NAME: Cluster name
 #    GCS_BUCKET: GCS bucket url for Dataproc resources (init actions)
 #    HARNESS_IMAGES_TO_PULL: Urls to SDK Harness' images to pull on dataproc workers (optional: 0, 1 or multiple urls for every harness image)
@@ -46,7 +46,7 @@ set -Eeuxo pipefail
 
 # GCloud properties
 GCLOUD_ZONE="${GCLOUD_ZONE:=us-central1-a}"
-DATAPROC_VERSION="${DATAPROC_VERSION:=2.0}"
+DATAPROC_VERSION="${DATAPROC_VERSION:=preview-debian11}"
 GCLOUD_REGION="us-central1"
 
 MASTER_NAME="$CLUSTER_NAME-m"
@@ -79,9 +79,9 @@ function get_leader() {
   while read line; do
     echo $line
     application_ids[$i]=`echo $line | sed "s/ .*//"`
-    application_masters[$i]=`echo $line | sed "s/.*$CLUSTER_NAME/$CLUSTER_NAME/" | sed "s/ .*//"`
+    application_masters[$i]=`echo $line | sed -E "s#.*(https?://)##" | sed "s/ .*//"`
     i=$((i+1))
-  done <<< $(gcloud compute ssh --zone=$GCLOUD_ZONE --quiet yarn@$MASTER_NAME --command="yarn application -list" | grep "$CLUSTER_NAME")
+  done <<< $(gcloud compute ssh --zone=$GCLOUD_ZONE --quiet yarn@$MASTER_NAME --command="yarn application -list" | grep "Apache Flink")
 
   if [ $i != 1 ]; then
     echo "Multiple applications found. Make sure that only 1 application is running on the cluster."
@@ -129,12 +129,11 @@ function create_cluster() {
   local image_version=$DATAPROC_VERSION
   echo "Starting dataproc cluster. Dataproc version: $image_version"
 
-  # Create one extra Dataproc VM for Flink's Job Manager
-  local num_dataproc_workers="$(($FLINK_NUM_WORKERS + 1))"
-
   # Docker init action restarts yarn so we need to start yarn session after this restart happens.
   # This is why flink init action is invoked last.
-  gcloud dataproc clusters create $CLUSTER_NAME --region=$GCLOUD_REGION --num-workers=$num_dataproc_workers  --metadata "${metadata}", --image-version=$image_version --zone=$GCLOUD_ZONE  --optional-components=FLINK,DOCKER  --quiet
+  # TODO(11/11/2022) remove --worker-machine-type=n1-standard-2 once N2 CPUs quota relaxed
+  # Dataproc 2.1 uses n2-standard-2 by default but there is N2 CPUs=24 quota limit
+  gcloud dataproc clusters create $CLUSTER_NAME --region=$GCLOUD_REGION --num-workers=$FLINK_NUM_WORKERS --worker-machine-type=n1-standard-2 --metadata "${metadata}", --image-version=$image_version --zone=$GCLOUD_ZONE  --optional-components=FLINK,DOCKER  --quiet
 }
 
 # Runs init actions for Docker, Portability framework (Beam) and Flink cluster

--- a/.test-infra/dataproc/flink_cluster.sh
+++ b/.test-infra/dataproc/flink_cluster.sh
@@ -46,8 +46,9 @@ set -Eeuxo pipefail
 
 # GCloud properties
 GCLOUD_ZONE="${GCLOUD_ZONE:=us-central1-a}"
+# TODO: replace preview once dataproc 2.1 released
 DATAPROC_VERSION="${DATAPROC_VERSION:=preview-debian11}"
-GCLOUD_REGION="us-central1"
+GCLOUD_REGION=`echo $GCLOUD_ZONE | sed -E "s/(-[a-z])?$//"`
 
 MASTER_NAME="$CLUSTER_NAME-m"
 
@@ -131,9 +132,11 @@ function create_cluster() {
 
   # Docker init action restarts yarn so we need to start yarn session after this restart happens.
   # This is why flink init action is invoked last.
-  # TODO(11/11/2022) remove --worker-machine-type=n1-standard-2 once N2 CPUs quota relaxed
+  # TODO(11/11/2022) remove --worker-machine-type and --master-machine-type once N2 CPUs quota relaxed
   # Dataproc 2.1 uses n2-standard-2 by default but there is N2 CPUs=24 quota limit
-  gcloud dataproc clusters create $CLUSTER_NAME --region=$GCLOUD_REGION --num-workers=$FLINK_NUM_WORKERS --worker-machine-type=n1-standard-2 --metadata "${metadata}", --image-version=$image_version --zone=$GCLOUD_ZONE  --optional-components=FLINK,DOCKER  --quiet
+  gcloud dataproc clusters create $CLUSTER_NAME --region=$GCLOUD_REGION --num-workers=$FLINK_NUM_WORKERS \
+  --master-machine-type=n1-standard-2 --worker-machine-type=n1-standard-2 --metadata "${metadata}", \
+  --image-version=$image_version --zone=$GCLOUD_ZONE  --optional-components=FLINK,DOCKER  --quiet
 }
 
 # Runs init actions for Docker, Portability framework (Beam) and Flink cluster

--- a/.test-infra/jenkins/CommonTestProperties.groovy
+++ b/.test-infra/jenkins/CommonTestProperties.groovy
@@ -26,7 +26,7 @@ class CommonTestProperties {
   }
 
   static String getFlinkVersion() {
-    return "1.13"
+    return "1.15"
   }
 
   static String getSparkVersion() {

--- a/.test-infra/jenkins/Flink.groovy
+++ b/.test-infra/jenkins/Flink.groovy
@@ -17,7 +17,7 @@
  */
 
 class Flink {
-  private static final String flinkDownloadUrl = 'https://archive.apache.org/dist/flink/flink-1.13.6/flink-1.13.6-bin-scala_2.12.tgz'
+  private static final String flinkDownloadUrl = 'https://archive.apache.org/dist/flink/flink-1.15.0/flink-1.15.0-bin-scala_2.12.tgz'
   private static final String hadoopDownloadUrl = 'https://repo.maven.apache.org/maven2/org/apache/flink/flink-shaded-hadoop-2-uber/2.8.3-10.0/flink-shaded-hadoop-2-uber-2.8.3-10.0.jar'
   private static final String FLINK_DIR = '"$WORKSPACE/src/.test-infra/dataproc"'
   private static final String FLINK_SCRIPT = 'flink_cluster.sh'

--- a/.test-infra/jenkins/job_LoadTests_Combine_Flink_Go.groovy
+++ b/.test-infra/jenkins/job_LoadTests_Combine_Flink_Go.groovy
@@ -107,7 +107,7 @@ def loadTestJob = { scope, triggeringContext, mode ->
         GO_SDK_CONTAINER
       ],
       initialParallelism,
-      "${DOCKER_CONTAINER_REGISTRY}/beam_flink1.13_job_server:latest")
+      "${DOCKER_CONTAINER_REGISTRY}/beam_flink${CommonTestProperties.getFlinkVersion()}_job_server:latest")
 
   // Execute all scenarios connected with initial parallelism.
   loadTestsBuilder.loadTests(scope, CommonTestProperties.SDK.GO, initialScenarios, 'combine', mode)

--- a/.test-infra/jenkins/job_LoadTests_Combine_Flink_Python.groovy
+++ b/.test-infra/jenkins/job_LoadTests_Combine_Flink_Python.groovy
@@ -132,7 +132,7 @@ def loadTestJob = { scope, triggeringContext, mode ->
         "${DOCKER_CONTAINER_REGISTRY}/${DOCKER_BEAM_SDK_IMAGE}"
       ],
       initialParallelism,
-      "${DOCKER_CONTAINER_REGISTRY}/beam_flink1.13_job_server:latest")
+      "${DOCKER_CONTAINER_REGISTRY}/beam_flink${CommonTestProperties.getFlinkVersion()}_job_server:latest")
 
   // Execute all scenarios connected with initial parallelism.
   loadTestsBuilder.loadTests(scope, CommonTestProperties.SDK.PYTHON, initialScenarios, 'Combine', mode)

--- a/.test-infra/jenkins/job_LoadTests_GBK_Flink_Go.groovy
+++ b/.test-infra/jenkins/job_LoadTests_GBK_Flink_Go.groovy
@@ -199,7 +199,7 @@ def loadTestJob = { scope, triggeringContext, mode ->
         GO_SDK_CONTAINER
       ],
       initialParallelism,
-      "${DOCKER_CONTAINER_REGISTRY}/beam_flink1.13_job_server:latest")
+      "${DOCKER_CONTAINER_REGISTRY}/beam_flink${CommonTestProperties.getFlinkVersion()}_job_server:latest")
 
   // Execute all scenarios connected with initial parallelism.
   loadTestsBuilder.loadTests(scope, CommonTestProperties.SDK.GO, initialScenarios, 'group_by_key', mode)

--- a/.test-infra/jenkins/job_LoadTests_GBK_Flink_Python.groovy
+++ b/.test-infra/jenkins/job_LoadTests_GBK_Flink_Python.groovy
@@ -146,7 +146,7 @@ def loadTest = { scope, triggeringContext ->
         "${DOCKER_CONTAINER_REGISTRY}/${DOCKER_BEAM_SDK_IMAGE}"
       ],
       numberOfWorkers,
-      "${DOCKER_CONTAINER_REGISTRY}/beam_flink1.13_job_server:latest")
+      "${DOCKER_CONTAINER_REGISTRY}/beam_flink${CommonTestProperties.getFlinkVersion()}_job_server:latest")
 
   def configurations = testScenarios.findAll { it.pipelineOptions?.parallelism?.value == numberOfWorkers }
   loadTestsBuilder.loadTests(scope, sdk, configurations, "GBK", "batch")

--- a/.test-infra/jenkins/job_LoadTests_ParDo_Flink_Go.groovy
+++ b/.test-infra/jenkins/job_LoadTests_ParDo_Flink_Go.groovy
@@ -127,7 +127,7 @@ def loadTestJob = { scope, triggeringContext, mode ->
         GO_SDK_CONTAINER
       ],
       numberOfWorkers,
-      "${DOCKER_CONTAINER_REGISTRY}/beam_flink1.13_job_server:latest")
+      "${DOCKER_CONTAINER_REGISTRY}/beam_flink${CommonTestProperties.getFlinkVersion()}_job_server:latest")
 
   loadTestsBuilder.loadTests(scope, CommonTestProperties.SDK.GO, batchScenarios(), 'ParDo', mode)
 }

--- a/.test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy
+++ b/.test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy
@@ -320,7 +320,7 @@ def loadTestJob = { scope, triggeringContext, mode ->
         "${DOCKER_CONTAINER_REGISTRY}/${DOCKER_BEAM_SDK_IMAGE}"
       ],
       numberOfWorkers,
-      "${DOCKER_CONTAINER_REGISTRY}/beam_flink1.13_job_server:latest")
+      "${DOCKER_CONTAINER_REGISTRY}/beam_flink${CommonTestProperties.getFlinkVersion()}_job_server:latest")
 
   loadTestsBuilder.loadTests(scope, CommonTestProperties.SDK.PYTHON, testScenarios, 'ParDo', mode)
 }

--- a/.test-infra/jenkins/job_LoadTests_SideInput_Flink_Go.groovy
+++ b/.test-infra/jenkins/job_LoadTests_SideInput_Flink_Go.groovy
@@ -79,7 +79,7 @@ def loadTestJob = { scope, triggeringContext, mode ->
         GO_SDK_CONTAINER
       ],
       numberOfWorkers,
-      "${DOCKER_CONTAINER_REGISTRY}/beam_flink1.13_job_server:latest")
+      "${DOCKER_CONTAINER_REGISTRY}/beam_flink${CommonTestProperties.getFlinkVersion()}_job_server:latest")
 
   loadTestsBuilder.loadTests(scope, CommonTestProperties.SDK.GO,
       batchScenarios(), 'SideInput', mode)

--- a/.test-infra/jenkins/job_LoadTests_coGBK_Flink_Go.groovy
+++ b/.test-infra/jenkins/job_LoadTests_coGBK_Flink_Go.groovy
@@ -159,7 +159,7 @@ def loadTestJob = { scope, triggeringContext, mode ->
         GO_SDK_CONTAINER
       ],
       numberOfWorkers,
-      "${DOCKER_CONTAINER_REGISTRY}/beam_flink1.13_job_server:latest")
+      "${DOCKER_CONTAINER_REGISTRY}/beam_flink${CommonTestProperties.getFlinkVersion()}_job_server:latest")
 
   loadTestsBuilder.loadTests(scope, CommonTestProperties.SDK.GO, batchScenarios(), 'CoGBK', mode)
 }

--- a/.test-infra/jenkins/job_LoadTests_coGBK_Flink_Python.groovy
+++ b/.test-infra/jenkins/job_LoadTests_coGBK_Flink_Python.groovy
@@ -137,7 +137,7 @@ def loadTest = { scope, triggeringContext ->
         "${DOCKER_CONTAINER_REGISTRY}/${DOCKER_BEAM_SDK_IMAGE}"
       ],
       numberOfWorkers,
-      "${DOCKER_CONTAINER_REGISTRY}/beam_flink1.13_job_server:latest")
+      "${DOCKER_CONTAINER_REGISTRY}/beam_flink${CommonTestProperties.getFlinkVersion()}_job_server:latest")
 
   loadTestsBuilder.loadTests(scope, CommonTestProperties.SDK.PYTHON, testScenarios, 'CoGBK', 'batch')
 }

--- a/.test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Flink.groovy
+++ b/.test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Flink.groovy
@@ -38,7 +38,7 @@ def chicagoTaxiJob = { scope ->
         "${DOCKER_CONTAINER_REGISTRY}/${beamSdkDockerImage}"
       ],
       numberOfWorkers,
-      "${DOCKER_CONTAINER_REGISTRY}/beam_flink1.13_job_server:latest")
+      "${DOCKER_CONTAINER_REGISTRY}/beam_flink${CommonTestProperties.getFlinkVersion()}_job_server:latest")
 
   def pipelineOptions = [
     parallelism             : numberOfWorkers,

